bindings_node/src/mls_client.rs

```diff
@@ -0,0 +1,209 @@
+use std::collections::HashMap;
+use std::ops::Deref;
+use std::sync::Arc;
+
+use crate::conversations::NapiConversations;
+use napi::bindgen_prelude::{Error, Result, Uint8Array};
+use napi_derive::napi;
+use xmtp_api_grpc::grpc_api_helper::Client as TonicApiClient;
+use xmtp_cryptography::signature::ed25519_public_key_to_address;
+use xmtp_id::associations::{Erc1271Signature, RecoverableEcdsaSignature};
+use xmtp_mls::builder::ClientBuilder;
+use xmtp_mls::identity::IdentityStrategy;
+use xmtp_mls::storage::{EncryptedMessageStore, EncryptionKey, StorageOption};
+use xmtp_mls::Client as MlsClient;
+
+pub type NapiXmtpClient = MlsClient<TonicApiClient>;
+
+#[napi]
+pub struct NapiClient {
+  inner_client: Arc<NapiXmtpClient>,
```

Comment:
is this `Arc` wrapper necessary?

Reply:
We added the `Arc` in the FFI bindings because Uniffi requires objects passed across the boundary to be an `Arc`. It seems `Napi` has a different way of handling this (maybe they're already wrapping it in Arc internally within their macro?)

My suggestion would be to just remove it and see if it compiles - if it compiles, it should be okay.

Reply:
when i remove the `Arc` wrapper, i can no longer `.clone()` the `inner_client`. how do i get around this without deriving the `Clone` trait on `xmtp_mls::Client`? passing by reference seems to bring up lifetime concerns.

Reply:
Ah, so that answers the question then - we need to share a reference to the `inner_client` in multiple places in Rust, but we don't want to handle lifetimes so we use an `Arc'. It's a Rust issue rather than a bindings issue.

Technically, after Andrew's refactor, I think we could refactor all of the other structs (like `FfiGroup`) to not need to hold that client reference, hence removing the need to clone it. But honestly I don't think it matters that much if it works today.

Reply:
Let me know if I didn't describe that clearly - happy to give more context on Rust stuff as I think it'll be useful going forward

Reply:
yep, great explanation! my limited understanding of `Arc` fits that. definitely curious about the refactor and how a reference would no longer be needed.

Reply:
I looked at it again, and I don't think a refactor would be helpful here! You don't need to spend a lot of time on this, but Andrew's refactor is here: https://github.com/xmtp/libxmtp/pull/727/files

Basically, an option could be to make the method signatures of `NapiGroup` match the method signatures of `MlsGroup`, where only the `context()` is passed in on construction, and the client is passed in by the caller on every method that needs it. But that requires the Node caller to manually specify the client every time it is needed, which might be inconvenient.

This sounds more complex than it is, and isn't super helpful, so don't worry too much about it!